core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -105,5 +105,16 @@ abstract class SparkFunSuite
       logInfo(s"\n\n===== FINISHED $shortSuiteName: '$testName' =====\n")
     }
   }
-
+  /**
+   * Creates a temporary directory, which is then passed to `f` and will be deleted after `f`
+   * returns.
+   *
+   * @todo Probably this method should be moved to a more general place
+   */
+  protected def withCreateTempDir(f: File => Unit): Unit = {
+    val dir = Utils.createTempDir()
```

Comment:
Is it better to call `.getCanonicalFile`, too?

Reply:
I'm not sure if I need call `.getCanonicalFile` again. i feel it's a little redundant. review `Utils.createTempDir()` --> `createDirectory` --> `dir.getCanonicalFile`.
It has been called `.getCanonicalFile`. thanks.

Reply:
Yes shouldn't be necessary here.